packages/members-api/lib/migrations/index.js

```diff
@@ -480,4 +480,21 @@ module.exports = class StripeMigrations {
             id: portalPlansSetting.id
         });
     }
+
+    async removeInvalidSubscriptions() {
+        const subscriptionModels = await this._StripeCustomerSubscription.findAll({
+            withRelated: ['stripePrice']
+        });
+        const invalidSubscriptions = subscriptionModels.filter((sub) => {
+            return !sub.toJSON().price;
+        });
+        if (invalidSubscriptions.length > 0) {
+            this._logging.warn(`Deleting ${invalidSubscriptions.length} invalid subscriptions`);
+        }
+        const deleteSubscriptionPromises = invalidSubscriptions.map((sub) => {
```

Comment:
Small thing, but are you aware that this will fire off multiple calls to the database simultaneously? Not sure the number of subscriptions we have to delete or if this will be an issue - but wanted to make sure this is a conscious decision!

If not we could use async/await to run each delete query in sequence - or we could do something like:

```js
const ids = invalidSubscriptions.map(sub => sub.id);
knex('members_stripe_customers_subscriptions').whereIn('id', ids).del();
```

Reply:
It was a conscious decision, but I its a fair point about it potentially causing issues with if we have large number of subscriptions to delete 🤔 I'll refactor it out to sequential deletion so its on safer side, as it won't anyways make much difference if invalid subscriptions are not too many(which is what i expect), but can better tackle any edge case

Reply:
@allouis Updated now, thanks :)